src/main/java/bio/terra/common/sam/SamRetry.java

```diff
@@ -90,7 +91,11 @@ public static void retry(SamVoidFunction function, Duration timeout)
   }
 
   private boolean isRetryable(ApiException apiException) {
-    return (apiException.getCode() == HttpStatus.INTERNAL_SERVER_ERROR.value());
+    return apiException.getCode() == TIMEOUT_STATUS_CODE
+        || apiException.getCode() == HttpStatus.SC_INTERNAL_SERVER_ERROR
+        || apiException.getCode() == HttpStatus.SC_BAD_GATEWAY
+        || apiException.getCode() == HttpStatus.SC_SERVICE_UNAVAILABLE
```

Comment:
This one's a little dangerous because if Sam is truly overloaded, we're just going to retry hammer it, making the overload worse. Client side throttling would be a better solution, but there's nothing off-the-shelf to use here.
Consider not retrying this case, very long retry times available in a retry mode that can tolerate long delays (i.e. during a flight), or just being really mean to Sam and retrying anyway.

Reply:
My inclination is to just be mean here, in practice I generally see this returned when hitting a pod which is still starting up, which is definitely a retryable case. I don't whether Sam also throws this under high load though, when retrying would be harmful.

Reply:
SG, let's leave a comment about that choice.